tensorflow/core/kernels/lookup_table_op.cc

```diff
@@ -56,14 +56,19 @@ class MutableHashTableOfScalars final : public LookupInterface {
 
   Status Find(OpKernelContext* ctx, const Tensor& key, Tensor* value,
               const Tensor& default_value) override {
-    const V default_val = default_value.flat<V>()(0);
     const auto key_values = key.flat<K>();
     auto value_values = value->flat<V>();
+    const auto default_flat = default_value.flat<V>();
+
+    int64 total = value_values.size();
+    int64 default_total = default_flat.size();
+    bool is_full_default = (total == default_total);
 
     tf_shared_lock l(mu_);
     for (int64 i = 0; i < key_values.size(); ++i) {
       value_values(i) = gtl::FindWithDefault(
-          table_, SubtleMustCopyIfIntegral(key_values(i)), default_val);
+          table_, SubtleMustCopyIfIntegral(key_values(i)),
+          is_full_default ? default_flat(i) : default_flat(0));
```

Comment:
So if is_full_default == false, we expect default_total = 1 i.e. a single scalar?

Reply:
> So if is_full_default == false, we expect default_total = 1 i.e. a single scalar?
Yes, Maybe you can pay attention to this line [`const auto default_flat = default_value.flat<V>();`](https://github.com/tensorflow/tensorflow/pull/43269/files/cb293ecd25239863aec8055d19551c696bf8d531#diff-4735359f29cfeb31e08b9f66b3121d4249e6db308138c6d6153a60f1403a4085R61)
Size of `dynamic_default_values` should be equal to `self._default_value.size()` or `keys.size() * self._default_value.size()`.  In the latter case, we can set different default values for each keys.

Reply:
Please add comments describing this logic.

Reply:
Done.